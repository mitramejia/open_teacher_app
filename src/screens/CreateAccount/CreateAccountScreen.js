// =============================================================
//
// Name: StudentHomeScreen
// -> Description:
//
// Author: mitramejia
// Created at: 8/21/17
//
// =============================================================

import React from 'react';
import { AppLoading } from 'expo';
import { Content, Button, Text, Item, Toast } from 'native-base';
import { Keyboard, TextInput, AsyncStorage } from 'react-native';
import { compose, gql, graphql, withApollo } from 'react-apollo';
import _ from 'lodash';
import propTypes from 'prop-types';
import {
  Form,
  Separator,
  InputField,
  LinkField,
  SwitchField,
  PickerField,
  DatePickerField,
  TimePickerField,
} from 'react-native-form-generator';
import { Ionicons } from '@expo/vector-icons';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import style from './style';
import { createAccount, signinUser, userQuery } from '../../api/graphql/queries';

class CreateAccountScreen extends React.Component {
  constructor(...props) {
    super(...props);
    this.state = {
      showToast: false,
      formData: {
        firstName: '',
        lastName: '',
        email: '',
        password: '',
      },
      formError: true,
    };
  }

  static navigationOptions = {
    title: 'Crear Cuenta',
  };

  _handleFormChange(formData) {
    this.setState({ formData });
    this.props.onFormChange && this.props.onFormChange(formData);
  }

  _createAccount() {
    const { firstName, lastName, email, password } = this.state.formData;
    this.props
      .createAccount({ variables: { firstName, lastName, email, password } })
      .then(response => {
        console.log(response);
        this.props
          .signinUser({ variables: { email, password } })
          .then(response => {
            console.log(response);
            AsyncStorage.setItem('token', response.data.signinUser.token);
            this.props.navigation.navigate('Home');
            Toast.show({
              text: `Hola ${firstName}`,
              position: 'bottom',
              buttonText: 'Ocultar',
              type: 'success',
              duration: 4000,
            });
          })
          .catch(e => {
            console.error(e.message);
            this.props.navigation.navigate('Home');
          });
      })
      .catch(e => {
        console.error(e.message);
        this.props.navigation.navigate('Home');
      });
    console.log('User: ' + this.props.data.user);
  }

  _checkAllFieldsAreValid() {
    return _.every(this.state.formData, value => {
      // Check all form fields aren't empty or undefined
      if (value !== '' || value !== undefined) {
        return true;
      }
    });
  }

  render() {
    if (this.props.data.loading) {
      return <AppLoading />;
    } else {
      return (
        <KeyboardAwareScrollView
          contentContainerStyle={style.container}
          keyboardShouldPersistTaps="always"
          ref="scroll">
          <Form
            ref="registrationForm"
            onChange={this._handleFormChange.bind(this)}
            label="Create Account">
            <InputField style={style.inputField} ref="firstName" label="Nombre" />
            <InputField style={style.inputField} ref="lastName" label="Apellido" />
            <InputField
              style={style.inputField}
              ref="email"
              label="Email"
              keyboardType="email-address"
            />
            <InputField
              style={style.inputField}
              ref="password"
              label="ContraseÃ±a"
              secureTextEntry={true}
            />
          </Form>
          {this._checkAllFieldsAreValid() &&
            <Button
              rounded
              block
              style={style.submitButton}
              onPress={this._createAccount.bind(this)}>
              <Text>Crear Cuenta</Text>
            </Button>}
          <Text>
            {JSON.stringify(this.state.formData)}
          </Text>
        </KeyboardAwareScrollView>
      );
    }
  }
}

CreateAccountScreen.propTypes = {
  client: React.PropTypes.object.isRequired,
  navigation: React.PropTypes.object.isRequired,
  createAccount: React.PropTypes.func.isRequired,
  signinUser: React.PropTypes.func.isRequired,
  data: React.PropTypes.object.isRequired,
};

export default graphql(createAccount, { name: 'createAccount' })(
  graphql(userQuery, { options: { fetchPolicy: 'network-only' } })(
    graphql(signinUser, { name: 'signinUser' })(withApollo(CreateAccountScreen))
  )
);
