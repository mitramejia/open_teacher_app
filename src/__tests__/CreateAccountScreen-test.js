import CreateAccountScreen from '../screens/CreateAccount/CreateAccountScreen';
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

it('renders a Create Account Screen', () => {
  const tree = renderer.create(<CreateAccountScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
