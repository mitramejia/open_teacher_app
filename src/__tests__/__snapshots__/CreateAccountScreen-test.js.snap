// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Create Account Screen 1`] = `
<RCTScrollView
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps={false}
  showsVerticalScrollIndicator={true}
  style={undefined}
>
  <View>
    <View
      style={undefined}
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#C8C7CC",
              "borderBottomWidth": 1,
              "paddingTop": 35,
            },
            undefined,
          ]
        }
      />
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            onLayout={[Function]}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "justifyContent": "center",
                  "lineHeight": 32,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
            suppressHighlighting={true}
          >
            First Name
          </Text>
          <TextInput
            containerStyle={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
            fieldRef="first_name"
            helpText={undefined}
            inputStyle={
              Array [
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                Object {},
                Object {
                  "textAlign": "right",
                },
                undefined,
              ]
            }
            keyboardType={undefined}
            label="First Name"
            labelStyle={
              Array [
                Object {
                  "fontSize": 17,
                  "justifyContent": "center",
                  "lineHeight": 32,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
            onChange={[Function]}
            onFocus={[Function]}
            onValidation={[Function]}
            placeholder="First Name"
            style={
              Array [
                Array [
                  Object {
                    "paddingLeft": 10,
                    "paddingRight": 10,
                  },
                  Object {},
                  Object {
                    "textAlign": "right",
                  },
                  undefined,
                ],
                Object {
                  "height": 44,
                },
              ]
            }
            validationFunction={
              Array [
                [Function],
                [Function],
              ]
            }
            value={undefined}
            width={NaN}
          />
        </View>
      </View>
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ],
            ]
          }
        >
          <TextInput
            containerStyle={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
            fieldRef="last_name"
            inputStyle={
              Array [
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                Object {},
                Object {},
                undefined,
              ]
            }
            keyboardType={undefined}
            labelStyle={
              Array [
                Object {
                  "fontSize": 17,
                  "justifyContent": "center",
                  "lineHeight": 32,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
            onChange={[Function]}
            onFocus={[Function]}
            onValidation={[Function]}
            placeholder="Last Name"
            style={
              Array [
                Array [
                  Object {
                    "paddingLeft": 10,
                    "paddingRight": 10,
                  },
                  Object {},
                  Object {},
                  undefined,
                ],
                Object {
                  "height": 44,
                },
              ]
            }
            value={undefined}
            width={NaN}
          />
        </View>
      </View>
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ],
            ]
          }
        >
          <TextInput
            containerStyle={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 1,
                  "justifyContent": "center",
                },
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
                undefined,
              ]
            }
            fieldRef="other_input"
            helpText="this is an helpful text it can be also very very long and it will wrap"
            inputStyle={
              Array [
                Object {
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                Object {},
                Object {},
                undefined,
              ]
            }
            keyboardType={undefined}
            labelStyle={
              Array [
                Object {
                  "fontSize": 17,
                  "justifyContent": "center",
                  "lineHeight": 32,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
            multiline={false}
            onChange={[Function]}
            onFocus={[Function]}
            onValidation={[Function]}
            placeholder="Other Input"
            style={
              Array [
                Array [
                  Object {
                    "paddingLeft": 10,
                    "paddingRight": 10,
                  },
                  Object {},
                  Object {},
                  undefined,
                ],
                Object {
                  "height": 44,
                },
              ]
            }
            value={undefined}
            width={NaN}
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 25,
              "marginTop": 9,
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7a7a7a",
              }
            }
          >
            this is an helpful text it can be also very very long and it will wrap
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#C8C7CC",
              "borderBottomWidth": 1,
              "paddingTop": 35,
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    >
      {}
    </Text>
  </View>
</RCTScrollView>
`;
